registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/service/impl/PacketExternalStatusServiceImpl.java

```diff
@@ -0,0 +1,182 @@
+						.getRegistrationStatus(null, null, null,
+						syncRegistrationEntity.getWorkflowInstanceId());
+				if (internalRegistrationStatusDto != null) {
+					PacketExternalStatusDTO packetExternalStatusDTO = getExternalStatusForPacket(
+							internalRegistrationStatusDto);
+					packetExternalStatusDTO.setPacketId(syncRegistrationEntity.getPacketId());
+					packetExternalStatusDTOList.add(packetExternalStatusDTO);
+				}
+			}
+			List<SyncRegistrationEntity> packetIdsNotAvailableInRegistration = syncRegistrationEntityList
+					.stream().filter(
+							syncRegistrationEntity -> packetExternalStatusDTOList.stream()
+									.noneMatch(packetExternalStatusDTO -> packetExternalStatusDTO.getPacketId()
+											.equals(syncRegistrationEntity.getPacketId())))
+					.collect(Collectors.toList());
+			packetExternalStatusDTOList.addAll(
+					convertEntityListToDtoListAndGetExternalStatusForPacket(packetIdsNotAvailableInRegistration));
+		}
+
+		regProcLogger.debug(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.USERID.toString(), "",
+				"PacketExternalStatusServiceImpl::getByPacketIds()::exit");
+		return packetExternalStatusDTOList;
+	}
+
+	private PacketExternalStatusDTO getExternalStatusForPacket(
+			InternalRegistrationStatusDto internalRegistrationStatusDto) {
+		PacketExternalStatusDTO packetStatusDTO = new PacketExternalStatusDTO();
+		if (internalRegistrationStatusDto.getStatusCode() != null) {
+			PacketExternalStatusCode packetExternalStatusCode = getPacketExternalStatusCode(
+					internalRegistrationStatusDto);
+
+			String mappedValue = packetExternalStatusCode.toString();
+
+			packetStatusDTO.setStatusCode(mappedValue);
+		} else {
+			regProcLogger.error(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.REGISTRATIONID.toString(),
+					internalRegistrationStatusDto.getReferenceRegistrationId(),
+					PlatformErrorMessages.RPR_RGS_REGISTRATION_STATUS_NOT_EXIST.getMessage());
+		}
+		return packetStatusDTO;
+	}
+
+	private PacketExternalStatusCode getPacketExternalStatusCode(
+			InternalRegistrationStatusDto internalRegistrationStatusDto) {
+		PacketExternalStatusCode mappedValue = null;
+		String status = internalRegistrationStatusDto.getStatusCode();
+		if (transactionTypeCodesBeforeUploadingToObjectStore
+				.contains(internalRegistrationStatusDto.getLatestTransactionTypeCode())
+				|| transactionTypeCodesUploadingToObjectStore
+						.contains(internalRegistrationStatusDto.getLatestTransactionTypeCode())) {
+			if (status.equalsIgnoreCase(RegistrationStatusCode.PROCESSING.toString())
+					|| status.equalsIgnoreCase(RegistrationStatusCode.REPROCESS.toString())
+					|| status.equalsIgnoreCase(RegistrationStatusCode.RESUMABLE.toString())) {
+				if (transactionTypeCodeTimeBasesResendRequired
+						.contains(internalRegistrationStatusDto.getLatestTransactionTypeCode())) {
+					long timeElapsedinPacketreceiver = checkElapsedTime(internalRegistrationStatusDto);

```

Comment:
In, I to be caps
But do use the stage name in a variable name since the stages are configurable
you can call it timeElapsedInCurrentStage

Reply:
Fixed